pkg/httpserver/httpserver.go

```diff
@@ -61,6 +60,16 @@ func NewHTTPServer(probes []health.Probes, httpProbes []health.HTTPProbe, metric
 	}
 }
 
+//NewDebugServer creates a new API Server for Debug
+func NewDebugServer(debugServer debugger.DebugServer, apiPort int32) *DebugServer {
```

Comment:
This seems very ambiguous because the function name is `NewDebugServer` and it takes in a `DebugServer` as an argument.

Reply:
I agree but I think @shalier is just trying to make only changes that are necessary to make this work. How about we rename this function `NewDebugHttpServer` to make it less ambiguous for now and then tackle changes in the `debugger` package next?